pkg/tekton/metapipeline/metaclient/metaclient.go

```diff
@@ -0,0 +1,324 @@
+package metaclient
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/jenkins-x/jx/pkg/apps"
+	jxclient "github.com/jenkins-x/jx/pkg/client/clientset/versioned"
+	"github.com/jenkins-x/jx/pkg/config"
+	"github.com/jenkins-x/jx/pkg/gits"
+	"github.com/jenkins-x/jx/pkg/jenkinsfile"
+	"github.com/jenkins-x/jx/pkg/jxfactory"
+	"github.com/jenkins-x/jx/pkg/kube"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/prow"
+	"github.com/jenkins-x/jx/pkg/tekton"
+	"github.com/jenkins-x/jx/pkg/tekton/metapipeline"
+	"github.com/jenkins-x/jx/pkg/tekton/syntax"
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+	tektonclient "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	kubeclient "k8s.io/client-go/kubernetes"
+)
+
+const (
+	pullRefOptionName   = "pull-refs"
+	sourceURLOptionName = "source-url"
+
+	retryDuration      = time.Second * 30
+	defaultCheckoutDir = "source"
+)
+
+// MetaClient contains the arguments to create the meta pipeline
+type MetaClient struct {
```

Comment:
I would think _Client_ would be enough. No need to repeat _meta_. Also, I was hoping to lock the interface down more. In its current form, this is just what we have in `step create pipeline` without Cobra. I was thinking of creating an actual interface, adding a factory method and a default implementation. 

Instead of having this exported properties on the struct, I'd like to define a couple of functions like `createMetaPipelineCRDs` and `applyMetaPipleCRDs` which take as parameters the required input. Certain options are specific to the client instance, whereas others are the actual parameters needed for the CRD generation. I'd like to separate this and also discuss with @ccojocar what type of parameters we want to pass between Lighthouse and meta pipeline client. For example, do we really want to keep Prows PullRef syntax?

Reply:
@hferentschik feel free to keep refactoring - this was literally just the first step along the road to decouple lighthouse from the options/cobra stuff

Reply:
I've moved the MetaClient into metapipeline as `Client`